fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/OverworldBiomes.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.biomes.v1;
+
+import net.fabricmc.fabric.impl.biomes.InternalBiomeData;
+import net.minecraft.world.biome.Biome;
+
+/**
+ * API that exposes some internals of the minecraft default biome source for the overworld
+ */
+public final class OverworldBiomes {
+
+	private OverworldBiomes() {
+	}
+
+	/**
+	 * Adds the biome to the specified climate group, with the specified weight
+	 *
+	 * @param biome the biome to be added
+	 * @param climate the climate group whereto the biome is added
+	 * @param weight the weight of the entry. Vanilla weights for each biome in each climate are listed in the {@link OverworldClimate} javadoc.
+	 */
+	public static void addBaseBiome(Biome biome, OverworldClimate climate, double weight) {
+		InternalBiomeData.addOverworldBaseBiome(climate, biome, weight);
+	}
+
+	/**
+	 * Adds the biome as a hills variant of the parent biome, with the specified weight
+	 *
+	 * @param parent the biome to where the hills variant is added
+	 * @param hills the biome to be set as a hills variant
+	 * @param weight the weight of the entry. The weight in this method corrosponds to the amount of times it is added
+	 * to the list that is randomly selected from when a hills biome is chosen.
+	 */
+	public static void addHillsBiome(Biome parent, Biome hills, int weight) {
+		InternalBiomeData.addOverworldHillsBiome(parent, hills, weight);
+	}
+
+	/**
+	 * Adds the biome as a shore/beach biome for the parent biome, with the specified weight
+	 *
+	 * @param parent the base biome to where the shore biome is added
+	 * @param shore the biome to be added as a shore biome
+	 * @param weight the weight of this entry. The weight in this method corrosponds to the amount of times it is added
+	 * to the list that is randomly selected from when a shore biome is chosen.
+	 */
+	public static void addShoreBiome(Biome parent, Biome shore, int weight) {
```

Comment:
Likewise here: if we want to standardize on double weights, just use double weights/rarities everywhere and warn on a weight that doesn't cleanly map to a supported value (say, [WARN] [fabric-biomes] Rounded up weight for biome mymod:mybiome (0.45 -> 1))

Reply:
It was all int weights until prospector changed base biomes

Reply:
Yeah, I think it might be a cool idea to add rational numbers to api-base (with the fluid API in mind) and then use that for weights? Or just doubles. Or use doubles now and add RationalNumber variants later.

Reply:
in my opinion, these should stay as ints for two reasons:

1. It would complicate the code to switch to using non-whole weights
2. It may encourage users of the api to think that these weights have anything to do with the weights in addBaseBiome, when in fact addBaseBiome's weights are relative to vanilla, while addShore/Hills/EdgeBiome methods are relative to only other entries of the same parent.

Reply:
I agree that doubles or RationalNumber here would add needless complexity with no functional benefit.  To make proper use of the information would require a full redesign of the underlying vanilla weighting system.

Reply:
Actually, that's not why. Our implementation of this doesn't deal with vanilla biomes at all, just modded ones. That's actually why ints are easier, because we don't need <1 weights, people can just set theirs in whole numbers relative to other mods.